.github/workflows/npm-publish.yml

```diff
@@ -0,0 +1,32 @@
+# This workflow will run tests using node and then publish a package to GitHub Packages when a release is created
+# For more information see: https://help.github.com/actions/language-and-framework-guides/publishing-nodejs-packages
+
+name: Node.js Package
+
+on:
+  release:
+    types: [created]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+      - run: npm ci
+
+  publish-npm:
```

Comment:
Probably want to gate this through an environment

Reply:
can you describe what kind of check you are thinking here? do you mean deploying a sample project that references this library and doing some manual testing?

Reply:
No, sorry like having a manual step to push to npm but I guess if this is trigged off release it is fine